### PR TITLE
Foreign keyboard fix

### LINKED ISSUE #1425: Enter key behavior in chat composites
When I type a chat message, I press the Enter key to confirm the conversion in Japanese input, but if I do that in Chrome and Safari on the Mac, the message is sent.
In Japanese input, I have to press Enter several times to confirm the conversion, but I'm hoping that the Enter key after confirmation will send the message.
In Windows, it works as expected.

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What
adds isComposing check to input box

# Why
Allows for foreign keyboard inputs to compose multiple words on mac

Fixes #1425
# How Tested

Ran locally to ensure premature messages aren't sent.

### CODE REVIEW

File: packages/react-components/review/react-components.api.md [hunk 1]
```diff
@@ -4,8 +4,6 @@
 
 ```ts
 
-/// <reference types="react" />
```
Comment (alkwa-msft) [inline packages/react-components/review/react-components.api.md:7]: how come we are seeing these removals in this PR as well?

@PorterNan  any ideas here?

File: packages/react-components/src/components/InputBoxComponent.tsx [hunk 2]
```diff
@@ -95,6 +95,9 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
 
   const onTexFieldKeyDown = useCallback(
     (ev: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      if ((ev.nativeEvent as KeyboardEvent).isComposing) {
```
[the remark this answers is not in the document]
Comment (dmceachernmsft) [reply packages/react-components/src/components/InputBoxComponent.tsx:98]: im working on a little doc about this we can share, might be something nice for them to incorporate in their text boxes to ensure consistent behavior on all platforms.

[the remark this answers is not in the document]
Comment (dmceachernmsft) [reply packages/react-components/src/components/InputBoxComponent.tsx:98]: but testing with Edward this does seem to solve the problem.

Comment (JamesBurnside) [inline packages/react-components/src/components/InputBoxComponent.tsx:98]: Avoid `as` as much as possible as then typescript can't typeguard against misuse.
Looks like the `as` shouldn't be needed?:
image

**Review by JamesBurnside — APPROVED** [review]
Remove the changes to the api.md files and use of `as` -- otherwise this looks great! I had no idea this was a thing!

**Comment by JamesBurnside** [comment]
Github tip, updated your PR description to say `Fixes <issues number>`, fixes and fix are special keywords that then track this PR as the fix to this issue and auto closes the issue when PR completes:
image

File: packages/react-components/src/components/InputBoxComponent.tsx [hunk 3]
```diff
@@ -92,6 +92,9 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
 
   const onTexFieldKeyDown = useCallback(
     (ev: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      if (ev.nativeEvent.isComposing) {
```
Comment (prprabhu-ms) [inline packages/react-components/src/components/InputBoxComponent.tsx:95]: Please add a comment in the code here explaining why this is needed. It's very easy to forget the reason for such checks.

**Review by prprabhu-ms — APPROVED** [review]

**Pull request merged**